eva_sub_cli/validators/validator.py

```diff
@@ -522,8 +522,40 @@ def _collect_md5sum_to_metadata(self):
                 with open(self.metadata_json_post_validation, 'w') as open_file:
                     json.dump(json_data, open_file)
 
+    def get_vcf_fasta_analysis_mapping(self):
+        vcf_fasta_analysis_mapping = []
+        with open(self.mapping_file) as open_file:
+            reader = csv.DictReader(open_file, delimiter=',')
+            for row in reader:
+                vcf_fasta_analysis_mapping.append({'vcf_file': row['vcf'], 'fasta_file': row['fasta']})
+
+        if self.metadata_json_post_validation:
+            with open(self.metadata_json_post_validation) as open_file:
+                try:
+                    vcf_analysis_dict = {}
+                    json_data = json.load(open_file)
+                    if json_data.get('files', []):
+                        for file in json_data.get('files', []):
+                            if file.get('fileName', []) and file.get('analysisAlias', []):
+                                vcf_analysis_dict[file.get('fileName')] = file.get('analysisAlias')
+
+                    for vcf_fasta_mapping in vcf_fasta_analysis_mapping:
+                        vcf_file = vcf_fasta_mapping.get('vcf_file')
+                        if vcf_file in vcf_analysis_dict:
+                            vcf_fasta_mapping.update({'analysis': vcf_analysis_dict.get(vcf_file)})
+                        else:
+                            vcf_fasta_mapping.update({'analysis': 'Could not be linked'})
+
+                    return vcf_fasta_analysis_mapping
+                except Exception as e:
+                    self.error('Error building Validation Report : Error getting info from metadata file' + str(e))
+        else:
+            self.error('Error building validation report : Metadata file not present')
```

Comment:
I think this does almost exactly the same thing as [this logic](https://github.com/EBIvariation/eva-sub-cli/blob/main/bin/check_fasta_insdc.py#L142) in the fasta check. It should be possible to either extract and re-use that logic, or adjust the results from that check and then use those directly in the report (currently looks like [this](https://github.com/EBIvariation/eva-sub-cli/blob/main/tests/test_report.py#L104) - I think it just needs the vcf file listing).

If we're worried about the fasta check not being run but still wanting to include the summary, I guess code reuse is the best bet. But #38 should help make that one more reliable anyway.